plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -30,13 +33,30 @@
         }
         return new DomainPushdownResult(domain, Domain.all(domain.getType()));
     };
-    PredicatePushdownController PUSHDOWN_AND_KEEP = (session, domain) -> new DomainPushdownResult(
-            domain.simplify(getDomainCompactionThreshold(session)),
-            domain);
+
     PredicatePushdownController DISABLE_PUSHDOWN = (session, domain) -> new DomainPushdownResult(
             Domain.all(domain.getType()),
             domain);
 
+    PredicatePushdownController CASE_INSENSITIVE_CHARACTER_PUSHDOWN = (session, domain) -> {
+        checkArgument(
+                domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
+                "CASE_SENSITIVE_PUSHDOWN can be used only for chars and varchars");
+
+        if (domain.isOnlyNull()) {
+            return FULL_PUSHDOWN.apply(session, domain);
+        }
+
+        if (domain.getValues().isDiscreteSet()) {
+            return new DomainPushdownResult(
+                    domain.simplify(getDomainCompactionThreshold(session)),
+                    domain);
+        }
+
+        // case insensitive predicate pushdown could return incorrect results for operators like `!=`, `<` or `>`
+        return DISABLE_PUSHDOWN.apply(session, domain);
```

Comment:
As discussed offline, it may be reasonable to have a session toggle to unlock legacy unsafe, performant behavior -- especially for the cases where user knows it's safe (and we do not)

Reply:
Sure, let me do it as separate PR.